File: leetcode/medium/1272-remove-interval/1272-remove-interval.py
from typing import List


class Solution:
    def removeInterval(self, intervals: List[List[int]], toBeRemoved: List[int]) -> List[List[int]]:
        result = []

        for interval in intervals:
            if interval[1] <= toBeRemoved[0] or toBeRemoved[1] <= interval[0]:
                result.append(interval)
                continue

            if interval[0] < toBeRemoved[0]:
                result.append([interval[0], toBeRemoved[0]])

            if toBeRemoved[1] < interval[1]:
                result.append([toBeRemoved[1], interval[1]])

        return result

File: leetcode/medium/1272-remove-interval/1272-remove-interval.kt
class Solution {
    fun removeInterval(intervals: Array<IntArray>, toBeRemoved: IntArray): List<List<Int>> {
        var result = mutableListOf<List<Int>>()

        for (interval in intervals) {
            if (interval[1] <= toBeRemoved[0] || interval[0] >= toBeRemoved[1]) {
                result.add(listOf(interval[0], interval[1]))
            } else {
                if (interval[0] < toBeRemoved[0]) {
                    result.add(listOf(interval[0], toBeRemoved[0]))
                }
                if (interval[1] > toBeRemoved[1]) {
                    result.add(listOf(toBeRemoved[1], interval[1]))
                }
            }
        }

        return result
    }
}
